heron/uploaders/src/java/com/twitter/heron/uploader/http/HttpUploader.java

```diff
@@ -44,55 +48,57 @@
 public class HttpUploader implements IUploader {
   private static final Logger LOG = Logger.getLogger(HttpUploader.class.getName());
 
+  private static final String FILE = "file";
   private Config config;
   private String topologyPackageLocation;
   private HttpPost post;
 
   @Override
   @SuppressWarnings("HiddenField")
   public void initialize(Config config) {
+    String uploaderUri = HttpUploaderContext.getHeronUploaderHttpUri(config);
+    Preconditions.checkArgument(StringUtils.isNotBlank(uploaderUri),
```

Comment:
Thanks for adding these checks.  We need to add these checks all around in Heron.  We currently lack any sort of organized config validation

Reply:
@jerrypeng Yes, exactly. Thanks for review ;)

Reply:
I created a config validation system using annotations in storm back in the day:

https://github.com/apache/storm/blob/master/storm-client/src/jvm/org/apache/storm/Config.java

Perhaps worthwhile to think if its worth while to have in Heron.  Though in Storm all configs are centralized which is not the case for Heron